R/FeatureSelectionForward.R

```diff
@@ -0,0 +1,86 @@
+#' @title FeatureSelectionForward
+#'
+#' @description
+#' FeatureSelection child class to conduct forward search.
+#'
+#' @section Usage:
+#'  ```
+#' fs = FeatureSelectionForward$new()
+#' ```
+#' See [FeatureSelection] for a description of the interface.
+#'
+#' @section Arguments:
+#' * `pe` (`[PerformanceEvaluator]`).
+#' * `tm` (`[Terminator]`).
+#'
+#' @section Details:
+#' `$new()` creates a new object of class [FeatureSelectionForward].
+#' `$get_result()` Returns selected features in each step.
+#' The interface is described in [FeatureSelection].
+#'
+#' Each step is possibly executed in parallel via [mlr3::benchmark()]
+#'
+#' @name FeatureSelectionForward
+#' @family FeatureSelection
+#' @examples
+#' task = mlr3::mlr_tasks$get("pima")
+#' measures = mlr3::mlr_measures$mget(c("classif.acc"))
+#' task$measures = measures
+#' learner = mlr3::mlr_learners$get("classif.rpart")
+#' resampling = mlr3::mlr_resamplings$get("cv", param_vals = list(folds = 5L))
+#' pe = PerformanceEvaluator$new(task, learner, resampling)
+#' tm = TerminatorPerformanceStep$new(threshold = 0.01)
+#' fs = FeatureSelectionForward$new(pe, tm)
+#' fs$calculate()
+#' fs$get_result()
+NULL
+
+#' @export
+#' @include FeatureSelection.R
+
+FeatureSelectionForward = R6Class("FeatureSelectionRandom",
+   inherit = FeatureSelection,
+   public = list(
+      initialize = function(pe, tm, max_features = NA) {
+         if(is.na(max_features)) {
+            max_features = length(pe$task$feature_names)
+         }
+
+         super$initialize(id = "forward_selection", pe = pe, tm = tm,
+                          settings = list(max_features = checkmate::assert_numeric(max_features,
```

Comment:
Instead of `settings` we should use `param_set` here inheriting from _paradox_. Similar to how it is done in the Filter class.

Reply:
`self$param_set = assert_param_set(param_set)`